com.stansassets.scene-management/BuildConfigurator/Editor/BundleRuleBaseRevealed.cs

```diff
@@ -187,25 +187,42 @@ private void ConvertBundleNamesToGroupNames(AddressableAssetsBuildContext buildC
             if (group.HasSchema<BundledAssetGroupSchema>()) {
                 var schema = group.GetSchema<BundledAssetGroupSchema>();
                 List<AssetBundleBuild> bundleInputDefinitions = new List<AssetBundleBuild>();
+                IEnumerable<AddressableAssetEntry> entries = null;
 
+#if UNITY_2021_3_OR_NEWER
+                 entries = BuildScriptPackedMode.PrepGroupBundlePacking(group, bundleInputDefinitions, schema);
+#else
                 var modeType = Type.GetType("UnityEditor.AddressableAssets.Build.DataBuilders.BuildScriptPackedMode,Unity.Addressables.Editor.dll");
                 if (modeType != null) {
                     var methodInfo = modeType.GetMethod("PrepGroupBundlePacking", BindingFlags.NonPublic | BindingFlags.Static);
                     if (methodInfo != null) {
-                        var entries = (IEnumerable<AddressableAssetEntry>) methodInfo.Invoke(null,
+                        entries = (IEnumerable<AddressableAssetEntry>) methodInfo.Invoke(null,
                             new object[] { group, bundleInputDefinitions, schema.BundleMode });
-                        m_assetEntries.AddRange(entries);
-
-                        for (int i = 0; i < bundleInputDefinitions.Count; i++) {
-                            if (m_bundleToAssetGroup.ContainsKey(bundleInputDefinitions[i].assetBundleName))
-                                bundleInputDefinitions[i] = CreateUniqueBundle(bundleInputDefinitions[i]);
+                    }
+                    else {
+                        Debug.LogWarning("Calculation of cross-reference dependencies in scenes is failed. " +
+                                         "It means that all scene dependencies will be included into scene bundle. " + 
+                                         "Type \"UnityEditor.AddressableAssets.Build.DataBuilders.BuildScriptPackedMode\" not found!");
+                        return;
+                    }
+                }
+                else {
+                    Debug.LogWarning("Calculation of cross-reference dependencies in scenes is failed. " +
+                                     "It means that all scene dependencies will be included into scene bundle. " +
+                                     "Method \"UnityEditor.AddressableAssets.Build.DataBuilders.BuildScriptPackedMode.PrepGroupBundlePacking\" not found!");
+                    return;
+                }
+#endif
+                m_assetEntries.AddRange(entries);
```

Comment:
I guess we will have _**ArgumentNullException**_ exception here if pass entries as null here (in case when we have older Unity version than 2021.3 and **_Type_** of **_method_** will not be found).
Please make sure that out change will not break the build assembly even if fixing duplicated scene dependencies is failed.

Reply:
In that case we do "return"

Reply:
Ah, I see now. Return is a bad practice from my perspective, but it's okay here.